homework/index.js

```diff
@@ -30,18 +33,169 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const root = document.getElementById('root');
+
+  function createHeader() {
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    return header;
+  }
+  const header = createHeader();
+
+  function createOptions(repositoryList) {
+    const select = createAndAppend('select', header, {
+      id: 'selector',
+    });
+    for (let i = 0; i < repositoryList.length; i++) {
+      const repository = repositoryList[i];
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: i,
+      });
+    }
+  }
+
+  function createMainContainer() {
+    const mainContainer = createAndAppend('div', root, {
+      id: 'container',
+    });
+    return mainContainer;
+  }
+  const mainContainer = createMainContainer();
+
+  function createListContributor() {
```

Comment:
This function does not deal with the contributor list but with the table containing repository details. The name `createListContributor` does not match what this function is doing.

Reply:
I chance it as containerListContributor.

Reply:
No, I'm afraid you are not getting the point. Contributors play no role at all in this function. Therefore the 'contributors' should not be part of the function name.

Reply:
containerListFotContributor may be ok but it will be to long
just tbodyCondainer can be also?
About naming the things I am not good enough. I need to work more...

Reply:
You are still not getting my point. Again, this function has nothing to do with contributors. So the word `contributors` should not be part of the name or even be considered to be part of the name. A good name could, for instance, could be `createRepoTableContainer` or something like that. The table is used to display details about the repository and not about contributors.